plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookManager.php

```diff
@@ -64,6 +77,63 @@ public function register() {
 
 		// Customize webhook HTTP arguments for our topics.
 		add_filter( 'woocommerce_webhook_http_args', array( $this, 'customize_webhook_http_args' ), 10, 3 );
+
+		// When the webhook is delivered (or not), mark the first event as delivered.
+		add_action( 'woocommerce_webhook_delivery', array( $this, 'mark_first_event_delivered' ), 10, 5 );
+	}
+
+	/**
+	 * Create the webhook for Agentic Commerce Protocol.
+	 *
+	 * @return void
+	 */
+	public function create_webhook(): void {
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return;
+		}
+
+		$name_prefix  = 'ACP'; // Not translated on purpose, to make search more specific.
+		$data_store   = \WC_Data_Store::load( 'webhook' );
+		$webhooks_ids = $data_store->search_webhooks(
+			array(
+				'search' => $name_prefix,
+				'status' => 'active',
+				'limit'  => -1,
+			)
+		);
+
+		foreach ( $webhooks_ids as $webhook_id ) {
+			$webhook = wc_get_webhook( $webhook_id );
+			if ( self::WEBHOOK_TOPIC === $webhook->get_topic() ) {
+				// There is a correct webhook already.
+				return;
+			}
+		}
+
+		/**
+		 * Filter the delivery URL for Agentic webhooks.
+		 *
+		 * @since 10.3.0
+		 *
+		 * @param string $delivery_url Delivery URL.
+		 */
+		$delivery_url = apply_filters( 'woocommerce_agentic_webhook_delivery_url', 'https://acp.invalid' );
+
+		// Include the non-translated prefix (ACP) to allow searching for the webhook by name.
+		$name = sprintf(
+			// translators: %s: webhook name prefix (ACP).
+			__( '%s: Order Created or Updated', 'woocommerce' ),
+			$name_prefix
+		);
+
+		$webhook = new \WC_Webhook();
+		$webhook->set_name( $name );
+		$webhook->set_user_id( get_current_user_id() );
+		$webhook->set_topic( self::WEBHOOK_TOPIC );
+		$webhook->set_secret( wp_generate_password( 50, false ) ); // This will be ignored, but is required.
```

Comment:
Similar to above, I think that we should consider passing in the secret to create_webhook and then use that here.

There's a note that it gets ignored. But, I don't see where it's overridden. And I think that we need the secret for the spec.

Reply:
I thought it gets overwritten here:

https://github.com/woocommerce/woocommerce/blob/f713dbb197a3b2e32151e4b84783eb1eb860fa51/plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookManager.php#L309

Now I see it's an additional header. Anyhow, yes. We'll only create the webhook with the settings from the UI.

Reply:
[I'm not automatically creating webhooks anymore](https://github.com/woocommerce/woocommerce/pull/61417/commits/578465144577d49d736a8181549faf53fe13f7f5). Marking this thread as resolved.